haystack/nodes/retriever/dense.py

```diff
@@ -1072,11 +1072,18 @@ def retrieve_batch(
         if scale_score is None:
             scale_score = self.scale_score
 
-        query_embs: List[np.ndarray] = []
-        for batch in self._get_batches(queries=queries, batch_size=batch_size):
-            query_embs.extend(self.embed_queries(queries=batch))
+        # embed_queries is already batched within by batch_size
+        query_embs: np.ndarray = self.embed_queries(queries=queries)
```

Comment:
I had a look at `self.embed_queries` (or to be accurate, `self._get_predictions`), and it seems like at the end it does a numpy `concatenate` operation over the batches. 

https://github.com/deepset-ai/haystack/blob/2878c57645449e9a81f691f9765568f01fe23d90/haystack/nodes/retriever/dense.py#L1143

Therefore, are you sure the queries actually come out in batches from this call?

Reply:
That's exactly right, I don't think `query_embs` come out batched from `self.embed_queries`.

Reply:
Oh I see! Then to make it super-clear (and prevent me from misunderstanding it again in a few months :smile:) I would expand it just a bit. Something like:
```
# embed_queries is already batched within by batch_size, so no need to batch the input here
```